core/src/net/cidr/v4.rs

```diff
@@ -45,6 +58,34 @@ impl Default for Ipv4Cidr {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct Ipv4CidrIterator {
+    next: Option<u32>,
```

Comment:
Could you clarify why `next` is an Option but end is not? Can `next` ever be `None`?

Reply:
`next` is an option as it changes on each iteration and can be point to nothing ahead or None. `end` is statically defined. `next` can be None, and I'm about to push an assert on the tests to show finishing iterating through the cidr range.

Reply:
ah yeah I see you have defined `next` can be `None` when end has reached. It makes sense now, thnx